src/catalog/src/durable/persist.rs

```diff
@@ -1502,22 +1474,26 @@ impl DurableCatalogState for PersistCatalogState {
             .into_iter()
             .map(RustType::from_proto)
             .map_ok(|key: StorageUsageKey| key.metric);
+        let mut events = Vec::new();
         let mut expired = Vec::new();
 
         for event in storage_usage {
             let event = event?;
-            if u128::from(event.timestamp()) < cutoff_ts {
+            if u128::from(event.timestamp()) >= cutoff_ts {
+                events.push(event);
+            } else if retention_period.is_some() {
                 debug!("pruning storage event {event:?}");
                 expired.push(event);
             }
         }
 
-        let mut txn = self.transaction().await?;
-        txn.remove_storage_usage_events(expired);
-        let updates = txn.get_and_commit_op_updates();
-        txn.commit().await?;
+        if !self.is_read_only() {
+            let mut txn = self.transaction().await?;
+            txn.remove_storage_usage_events(expired);
+            txn.commit_internal().await?;
+        }
 
-        Ok(updates)
+        Ok(events)
```

Comment:
It doesn't look like we're sorting by ID anywhere? Or is that a property upheld by listing out of Persist?

Reply:
Thank you! I totally missed that. Just pushed a fix.